test/unit/globalInfo/accessors/aliasAccessorTest.ts

```diff
@@ -0,0 +1,149 @@
+  const username2 = 'foobar@salesforce.com';
+  const alias1 = 'MyAlias';
+  const alias2 = 'MyOtherAlias';
+  const alias3 = 'MyThirdAlias';
+  const org = { username: username1, orgId: '12345' };
+  const token = { token: '123', url: 'https://login.salesforce.com', user: username1 };
+  const $$ = testSetup();
+
+  beforeEach(async () => {
+    $$.SANDBOXES.CONFIG.restore();
+
+    stubMethod($$.SANDBOX, GlobalInfo.prototype, 'read').callsFake(async function (this: GlobalInfo) {
+      const contents = {
+        orgs: { [username1]: org },
+        aliases: {
+          [alias1]: username1,
+          [alias2]: username2,
+          [alias3]: username1,
+        },
+        tokens: { [username1]: token },
+      };
+      this.setContentsFromObject(contents);
+      return this.getContents();
+    });
+    stubMethod($$.SANDBOX, GlobalInfo.prototype, 'write').callsFake(() => {});
+  });
+
+  describe('getAll', () => {
+    it('should return all the aliases if no username is provided', async () => {
+      const globalInfo = await GlobalInfo.create();
+      const aliases = globalInfo.aliases.getAll();
+      expect(aliases).to.deep.equal({
+        [alias1]: username1,
+        [alias2]: username2,
+        [alias3]: username1,
+      });
+    });
+
+    it('should return all the aliases for a given username', async () => {
+      const globalInfo = await GlobalInfo.create();
+      const aliases = globalInfo.aliases.getAll(username1);
+      expect(aliases).to.deep.equal([alias1, alias3]);
+    });
+  });
+
+  describe('get', () => {
+    it('should return first alias of username', async () => {
+      const globalInfo = await GlobalInfo.create();
+      const aliases = globalInfo.aliases.get(username1);
+      expect(aliases).to.equal(alias1);
+    });
+  });
+
+  describe('getUsername', () => {
+    it('should return the username that corresponds to an alias', async () => {
+      const globalInfo = await GlobalInfo.create();
+      const username = globalInfo.aliases.getUsername(alias1);
+      expect(username).to.equal(username1);
+    });
+
+    it('should return null if the alias does not exist', async () => {
+      const globalInfo = await GlobalInfo.create();
+      const username = globalInfo.aliases.getUsername('DOES_NOT_EXIST');
+      expect(username).to.be.null;
+    });
+  });
+
+  describe('resolveUsername', () => {
+    it('should return a username if a username was provided', async () => {
+      const globalInfo = await GlobalInfo.create();
+      const username = globalInfo.aliases.resolveUsername(username1);
+      expect(username).to.equal(username1);
+    });
+
+    it('should return a username if an alias was provided', async () => {
+      const globalInfo = await GlobalInfo.create();
+      const username = globalInfo.aliases.getUsername(alias1);
+      expect(username).to.equal(username1);
+    });
+  });
+
+  describe('set', () => {
+    it('should set an alias for a username', async () => {
```

Comment:
@mdonnalley 

`should set alias to an alias` y=foobar@salesforce.com, a1=y, a2=a1 then getUsername('a2') = ?
`should not set an alias to itself` alias:set foobar=foobar (this is currently possible)

Reply:
Good point but there's never been any safeguard against aliasing something that isn't an org username. I'm not sure that we even want to enforce that since there may be some valid use cases in the future where we do want to create aliases for things that are orgs.